pkg/webhook/admission/validator.go

```diff
@@ -67,7 +69,13 @@ func (h *validatingHandler) Handle(ctx context.Context, req Request) Response {
 		}
 
 		err = obj.ValidateCreate()
+
 		if err != nil {
+
+			isStatusError, status := isStatusError(&err)
+			if isStatusError {
+				return ValidationResponseFromStatus(false, status)
```

Comment:
```suggestion
var statusError errors.StatusError
if goerrors.As(&statusError) {
  return ...
}
```

Reply:
(the other nice thing about this is that if the status error is wrapped in such a way that it's intended to be unwrapped, we'll still pull the status)

Reply:
is there any corresponding interface for status-returning errors in the k8s api errors package that we should be asserting on?  (basically, is there a pattern for "my custom error returns status information)